src/Encrypt/Openssl.php

```diff
@@ -140,7 +143,10 @@ protected function _setKeys($keys)
                         throw new Exception\InvalidArgumentException("Private key '{$cert}' not valid");
                     }
 
-                    openssl_free_key($test);
+                    // PHP 8 automatically frees the key instance and deprecates the function
+                    if (PHP_VERSION_ID < 80000) {
+                        openssl_free_key($test);
+                    }
```

Comment:
This is a duplication of code logic and therefore it should be extracted as a private method.